### PR TITLE
Reader: connect recommended sites

### PR DESCRIPTION
**Pull request by samouri**
Implement the recommendations from https://github.com/Automattic/wp-calypso/issues/12952

When the user has not entered a search term, two site recommendations should be shown:

**Wider viewports:**
manage-following- desktop

Recommendations should be formatted just like in-stream recommendations. Clicking Follow or Dismiss causes that site to disappear and a new one to appear.

### CODE REVIEW

File: client/reader/following-manage/index.jsx [hunk 1]
```diff
@@ -97,6 +103,7 @@ class FollowingManage extends Component {
 					<h1>{ translate( 'Manage Followed Sites' ) }</h1>
 				</MobileBackToSidebar>
 				{ searchResults.length === 0 && <QueryReaderFeedsSearch query={ sitesQuery } /> }
+				{ recommendedSites.length === 0 && <QueryReaderRecommendedSites seed={ this.state.seed } /> }
```
Comment (blowery) [inline client/reader/following-manage/index.jsx]: how do we differentiate between "we asked for recs and none were found" and "i have not yet asked for recs and need to"?

Comment (samouri) [inline client/reader/following-manage/index.jsx]: I didn't think none could be found so I didn't include the null state for unfetched...

**Comment by samouri** [comment]
depends on https://github.com/Automattic/wp-calypso/pull/13472

File: client/blocks/reader-recommended-sites/index.jsx [hunk 2]
```diff
@@ -49,7 +49,7 @@ export class RecommendedSites extends React.PureComponent {
 						map(
 							sites,
 							( site, index ) => {
-								const siteId = site.siteId;
+								const siteId = site.siteId || site.blogId;
```
Comment (bluefuton) [inline client/blocks/reader-recommended-sites/index.jsx]: Why do we have an unfortunate mixture of siteId and blogId here?

Comment (samouri) [inline client/blocks/reader-recommended-sites/index.jsx]: i could do a map higher up to turn blogId into siteId.  

I think ideally we could pick one and then change them all at the data-layer.

🤷‍♂️

Comment (bluefuton) [inline client/blocks/reader-recommended-sites/index.jsx]: That'd be nice. It'd be good to deal with one consistently throughout Reader, and we generally go with blogId.

They usually mean the same thing, except when they don't...

Comment (samouri) [inline client/blocks/reader-recommended-sites/index.jsx]: ok. blogId it is!

**Comment by bluefuton** [comment]
A nice-to-have: it'd be good if the replacement recommendation appeared in the same slot as the one I've dismissed. Currently if I dismiss the left one, the right one moves over.

**Comment by samouri** [comment]
I agree that it would better if we implemented in a way that let us slot them.  Would probably need to create a connected component like `RecommendedPosts`.

**Pull request merged**